frontend/test/pytest/test_peephole_optimizations.py

```diff
@@ -29,6 +29,229 @@
 #
 
 
+class TestHermitianGatesSelfInverse:
```

Comment:
Can we add a test for a case like `H-X-X-H`? i.e. nested Hermitian gates should be canceled too.

Reply:
same for lit test

Reply:
Discussed in person, we only need mlir lit tests.